rskj-core/src/test/java/co/rsk/peg/LegacyErpFederationTest.java

```diff
@@ -71,48 +75,54 @@ void setup() {
         standardKeys = Arrays.asList(
             federator0PublicKey, federator1PublicKey, federator2PublicKey,
             federator3PublicKey, federator4PublicKey, federator5PublicKey,
-            federator6PublicKey, federator7PublicKey, federator8PublicKey, federator9PublicKey
+            federator6PublicKey, federator7PublicKey, federator8PublicKey,
+            federator9PublicKey
         );
-
-        networkParameters = bridgeConstants.getBtcParams();
+        defaultThreshold = standardKeys.size() / 2 + 1;
         emergencyKeys = bridgeConstants.getErpFedPubKeysList();
+        emergencyThreshold = emergencyKeys.size() / 2 + 1;
         activationDelayValue = bridgeConstants.getErpFedActivationDelay();
+        networkParameters = bridgeConstants.getBtcParams();
         activations = mock(ActivationConfig.ForBlock.class);
         when(activations.isActive(ConsensusRule.RSKIP284)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
 
         federation = createDefaultLegacyErpFederation();
     }
 
-    private LegacyErpFederation createDefaultLegacyErpFederation() {
+    private ErpFederation createDefaultLegacyErpFederation() {
         List<FederationMember> standardMembers = FederationTestUtils.getFederationMembersWithBtcKeys(standardKeys);
         Instant creationTime = ZonedDateTime.parse("2017-06-10T02:30:00Z").toInstant();
         long creationBlockNumber = 0L;
+        erpRedeemScriptBuilder =
+            NonStandardErpRedeemScriptBuilderFactory.defineNonStandardErpRedeemScriptBuilder(activations, networkParameters);
```

Comment:
If RSKIP293 is always true in these tests then there's no point in calling the factory, right?

Reply:
but is not always true, in fact, some tests test things when rskip293 and/or rskip284 are not activated, and the idea is that we are testing that the builder is got correctly.
(by the way, this class changes a bit after adding the tests)